capstone/src/main/webapp/youtubeScript.js

```diff
@@ -0,0 +1,137 @@
+  const requestedAttributes = [];
+  for (let attribute of attributes) {
+    if (attribute.checked == true) {
+      requestedAttributes.push(attribute.value);	
+    }	
+  }
+  return requestedAttributes;
+}
+
+/** Calls the perspective API */
+async function callPerspective(text, lang, requestedAttribute) {
+  const response = await fetch('/call_perspective', {
+    method: 'POST',
+    headers: {'Content-Type': 'application/json',},
+    body: JSON.stringify({text: text, lang: lang, requestedAttributes: requestedAttribute})});
+  const toxicityData = await response.json();
+  return toxicityData.attributeScores[requestedAttribute].summaryScore.value;
+}
+
+/** Loads the Google Charts API */
+function loadChartsApi(toxicityData) {
+  google.charts.load('current', {'packages':['corechart']});
+  google.charts.setOnLoadCallback(function() {drawBarChart(toxicityData);}); 
+}
+
+/** Draws a Google BarChart from a map. */
+function drawBarChart(toxicityData) {
+  document.getElementById('chart-container').innerHTML = '';
+  const data = google.visualization.arrayToDataTable([[{label: 'Attribute'}, {label: 'Score', type: 'number'}, {role: "style"}]]);
+  for (let [attribute, attributeScoresAvg] of toxicityData) {
+    var color = '#6B8E23'; // Green
+    const score = attributeScoresAvg;
+    if (score >= 0.8) {
+      color = '#DC143C'; // Red
+    } else if (score >= 0.2) {
+      color = '#ffd800'; // Yellow
+    }
+    data.addRow([attribute , score, color]);
+  }
+
+  data.sort({column: 1, desc: false});
+
+  const options = {
+    title: 'Perspective Feedback',
+    bars: 'horizontal',
+    height: 700,
+    legend: {position: "none"},
+    theme: 'material', 
+    hAxis: {viewWindow: {min: 0, max: 1}}
+  };
+
+  const chart = new google.visualization.BarChart(document.getElementById('chart-container'));
+  chart.draw(data, options);
+}
+
+/** Shows the avaiable attributes given a language selected on text analyzer page */
+function showAvailableAttributes() {
+  const langElement = document.getElementById('languageForAnalysis');
+  if (!langElement) {
+    return;
+  }
+  const lang = langElement.value;
+  const avaiableAttributesElement = document.getElementById('available-attributes');
+  avaiableAttributesElement.innerHTML = '';
+	
+  const attributes = ATTRIBUTES_BY_LANGUAGE[lang];
+  attributes.forEach(function(attribute) {
+    const checkbox = document.createElement('input');
+    checkbox.type = "checkbox";
+    checkbox.value = attribute;
+    checkbox.id = attribute + '-checkbox';
+    checkbox.checked = true;
+  
+    const label = document.createElement('label');
+    label.htmlFor = attribute + '-checkbox';
+    label.appendChild(document.createTextNode(attribute));
+  
+    avaiableAttributesElement.appendChild(checkbox);
+    avaiableAttributesElement.appendChild (document.createTextNode (" "));
+    avaiableAttributesElement.appendChild(label);
+    avaiableAttributesElement.appendChild (document.createTextNode (" "));
```

Comment:
nit: Let's remove the extra spaces between the function calls

Reply:
addressed in "address nits"